tests/smoke/alp_stedc.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+ *   Copyright 2022 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include <alp.hpp>
+#include <alp/algorithms/symm_tridiag_eigensolver.hpp>
+#include <graphblas/utils/iscomplex.hpp> // use from grb
+#ifdef DEBUG
+#include "../utils/print_alp_containers.hpp"
+#endif
+
+using namespace alp;
+
+using BaseScalarType = double;
+using Orthogonal = structures::Orthogonal;
+
+using ScalarType = BaseScalarType;
+
+// not fully implemented structures
+using HermitianOrSymmetricTridiagonal = structures::SymmetricTridiagonal;
+
+//fully implemented structures
+using HermitianOrSymmetric = structures::Symmetric;
+
+constexpr BaseScalarType tol = 1.e-5;
+constexpr size_t RNDSEED = 11235;
+
+//temp function untill (tridiagonal)Hermitian containter is implemented
+//** gnerate tridiagonal-symmetric-hermitian matrix in a rectangular container */
+template<
+	typename T
+>
+std::vector< T > generate_symmherm_tridiag_matrix_data(
+	size_t N,
+	const typename std::enable_if<
+		grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::vector< T > data( N * N );
+	std::fill(data.begin(), data.end(), static_cast< T >( 0 ) );
+	std::srand( RNDSEED );
```

Comment:
Similar comment as in the previous pull request: Due to srand call within the generate function, the function will produce the same data for every call. If it is called only once in the application, this is not a problem. Might be an issue when/if the function is reused somewhere else.